python/paddle/fluid/contrib/slim/quantization/quantization_pass.py

```diff
@@ -281,37 +473,65 @@ def _quant_preprocess(op_node):
 
         def _transform_forward(graph, op):
             op.op()._set_attr("quantization_type", "qat_with_weight")
-            for var_node in op.inputs:
+            inputs = op.inputs
+            for var_node in inputs:
                 if var_node.name() not in op.input_arg_names():
                     continue
                 if var_node.name() in dequantized_vars:
                     dequant_var_node = dequantized_vars[var_node.name()]
                 else:
+
+                    name = var_node.name()
+                    if name in processed_vars:
+                        continue
+
+                    if var_node.name() in persistable_vars:
+                        is_weight = True
+                    else:
+                        is_weight = False
+
+                    if is_weight and self._weight_preprocess_func is not None:
+                        var_node = self._insert_func(
+                            graph, self._weight_preprocess_func, var_node, op)
+                    if not is_weight and self._act_preprocess_func is not None:
+                        var_node = self._insert_func(
+                            graph, self._act_preprocess_func, var_node, op)
+
+                    if is_weight and self._weight_quantize_func is not None:
+                        target_out_node = self._insert_func(
+                            graph, self._weight_quantize_func, var_node, op)
+                        processed_vars[name] = target_out_node
+                        continue
+                    elif not is_weight and self._act_quantize_func is not None:
+                        target_out_node = self._insert_func(
+                            graph, self._act_quantize_func, var_node, op)
+                        processed_vars[name] = target_out_node
+                        continue
```

Comment:
If self._act_quantize_func is not None or self._weight_quantize_func is not None, it doesn't insert fake_quant op before the quantized op?

Reply:
if self._act_quantize_func is not None,  it doesn't insert  both fake_quant op and fake dequant op

Reply:
Got it